server/auth.go

```diff
@@ -447,8 +451,80 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 	if s.trustedKeys != nil {
 		if c.opts.JWT == _EMPTY_ {
 			s.mu.Unlock()
-			c.Debugf("Authentication requires a user JWT")
-			return false
+
+			var acc *Account
```

Comment:
If we could move this code into its own function (if possible, that is, don't need a lot of variables defined in processClientOrLeafAuthentication), that would be great. The function processClientOrLeafAuthentication is quite long already..

Reply:
fixed